lib/ansible/executor/process/connection.py

```diff
@@ -0,0 +1,204 @@
+                setattr(self.connection, '_connected', False)
+
+        display.debug('shutdown complete')
+
+    def start(self):
+        """ Called to initiate the connect to the remote device
+        """
+        play_context = self.connection._play_context
+
+        ssh = connection_loader.get('ssh', class_only=True)
+        cp = ssh._create_control_path(play_context.remote_addr, play_context.port, play_context.remote_user)
+
+        # create the persistent connection dir if need be and create the paths
+        # which we will be using later
+        tmp_path = unfrackpath(C.PERSISTENT_CONTROL_PATH_DIR)
+        makedirs_safe(tmp_path)
+
+        lock_path = unfrackpath("%s/.ansible_pc_lock" % tmp_path)
+        socket_path = unfrackpath(cp % dict(directory=tmp_path))
+
+        setattr(self.connection, '_socket_path', socket_path)
+
+        # if the socket file doesn't exist, spin up the daemon process
+        lock_fd = os.open(lock_path, os.O_RDWR | os.O_CREAT, 0o600)
+        fcntl.lockf(lock_fd, fcntl.LOCK_EX)
+
+        display.vvvv('socket_path is %s' % socket_path, host=play_context.remote_addr)
+
+        if not os.path.exists(socket_path):
+            display.vvvv('forking process to start persistent connection', host=play_context.remote_addr)
+            pid = fork_process(redirect_stdout=False, redirect_stderr=False)
+            if pid == 0:
+                rc = 0
+                try:
+                    self.start_connection()
+                    self.listen(socket_path)
+                except AnsibleError as exc:
+                    display.vvvv(str(exc), host=play_context.remote_addr)
+                    display.debug(traceback.format_exc())
+                    rc = 1
+                except Exception as exc:
+                    display.vvvv('failed to create control socket for host %s' % play_context.remote_addr, host=play_context.remote_addr)
+                    display.debug(traceback.format_exc())
+                    rc = 1
+                fcntl.lockf(lock_fd, fcntl.LOCK_UN)
+                os.close(lock_fd)
+                if rc == 0:
+                    self.run()
+                sys.exit(rc)
+        else:
+            display.vvvv('found existing socket, reusing it!', host=play_context.remote_addr)
+
+        # increase the connect timeout value by one to avoid any race conditions
+        timeout = C.PERSISTENT_CONNECT_TIMEOUT + 1
+        while bool(timeout):
+            if os.path.exists(socket_path):
+                elapsed = (C.PERSISTENT_CONNECT_TIMEOUT + 1) - timeout
+                display.vvvv('connected to local socket in %s second(s)' % elapsed, host=play_context.remote_addr)
+                setattr(self.connection, '_connected', True)
+                break
+            time.sleep(1)
+            timeout -= 1
+        else:
+            raise AnsibleError('timeout waiting for local socket', play_context.remote_addr)
+
```

Comment:
To handle privilege escalation related activities `update_play_context()` should be somewhere in this function?